### ISSUE TITLE
Ability to send notifications for new episodes of ongoing seasons

### ISSUE THREAD

**Issue by russjr08** [Fallenbagel/jellyseerr#480]
### Description

I've noticed when requesting shows with Jellyseer, that if I request a TV show season that has already aired all of its episodes, I'll get a notification when everything is downloaded as expected. However, when requesting a season that is still actively airing new episodes, you won't receive notifications until the full season has aired, and the last episode has been downloaded.

### Desired Behavior

It would be nice if it were possible to enable an option to get a notification that an episode was downloaded/imported, even if the full season hasn't been downloaded yet (because its still actively airing). Ideally though, Jellyseer would be able to *only* do this when a newly aired episode has just been imported (so that you don't request a season of say 26 episodes that have all aired, resulting in 26 notifications as they all import).

This could even be extended to have a "series partially available" notification option. As an example, lets say three weeks ago I requested Foundation (the last episode of the season only just aired this last week), once all episodes that had aired at that point were imported, a notification could be sent. After that point, it could send a notification each week when the next episode is downloaded.

### Code of Conduct

- [X] I agree to follow Overseerr's Code of Conduct

**Comment by ojmaster** [comment]
bump, i'd like this aswell

**Comment by martindacost** [comment]
Same here ! Would be cool to have notification when new episode comes out and it's available on the media server, like movies

**Comment by SpiderMan-ParkerTalley** [comment]
I'm definitely interested in something like this. A way to receive notifications about new content added from stuff on your watchlist would also be greatly appreciated.

**Comment by Vexz00** [comment]
I just started using Jellyseerr and am very baffled that this isn't even an option. I thought this would be the default. Without that feature it feels like this software only does 50% of what I want. I hope this feature will be added soon, but I have my doubts, considering how long this feature request has been open.

**Comment by fallenbagel** [comment]
> but I have my doubts, considering how long this feature request has been open.

If this was not a feature that is planned, the feature request would be closed as not planned. And if you check one message above, there already is a **draft** PR opened to implement this exact feature. Currently there are discussions on how to implement this properly so it does not end up bulk spam.

**Comment by Vexz00** [comment]
Fair point. At least Sonarr can send notifications for single file imports. Should be a good workaround for now.

**Comment by Sdhb1** [comment]
There seem to be many duplicates of this very request. Can someone confirm if it is on the roadmap? All requests seems to link back to one another. I'm not clear if this has or will be availible in jellyseer or overseer. Ty

**Comment by fallenbagel** [comment]
> There seem to be many duplicates of this very request. Can someone confirm if it is on the roadmap? All requests seems to link back to one another. I'm not clear if this has or will be availible in jellyseer or overseer. Ty

A duplicate issue that is closed and linked to another issue which is open (like this one) means that it's still in the roadmap. If it wasn't this will be closed as `not planned`.

**Comment by parkuman** [comment]
I was just thinking about how the multiple added notifications could get annoying for something like a Discord server with multiple users, each who may have different TV watching interests.

Say user1 requests a show, and week by week a notification gets sent to a shared Discord channel for each new episode. All the other users would get notified of this and it would get pretty annoying, especially for a show you have no interest in watching. 

I was thinking it could be cool to have an option to only notify the user who requested the show about each new episode drop. Maybe we could do this via a mapping in the Discord notification settings? i.e.

```
seerr username -> discord username
```

If multiple users wanted to be notified for each new episode drop, we'd need some sort of system for them to opt-in to notifications too.


Just a thought! I am really excited for this to be implemented as I'd use it with family and friends all the time.

**Comment by fallenbagel** [comment]
> I was thinking it could be cool to have an option to only notify the user who requested the show about each new episode drop. Maybe we could do this via a mapping in the Discord notification settings? i.e.\n\nseerr username -> discord username

This is already possible for requests? (Not episode but season wise). When you enable mentioned and you add in discord id in user settings > general.

**Comment by parkuman** [comment]
@fallenbagel You're totally right. I was digging in the General settings screen before writing the message rather than user settings. 😆 

Super useful, thank you!

**Comment by ghaisasadvait** [comment]
I vibecoded my own docker compose based app for this:

Behavior
On each poll:

- Load all approved/completed TV requests from Seerr (GET /api/v1/request?mediaType=tv)
- Track HD and 4K separately (state key is (tmdb_id, is4k))
- Keep shows whose media status is PARTIALLY_AVAILABLE (status / status4k as appropriate)
- Resolve Sonarr (from Seerr settings or manual Agent settings) using serviceId / externalServiceId (or 4K equivalents)
- Count Sonarr episodes with airDate <= today vs hasFile for the requested seasons
- Email once per request_id when downloaded == aired and that request has not been notified yet
- Skip fully AVAILABLE shows (Seerr’s own notification covers those)
- Later episode airings/downloads for the same request do not re-trigger email

Overlapping polls (scheduler + Check now) are serialized so the same request cannot be emailed twice in parallel.


Screenshot:

<img width="751" height="440" alt="Image" src="https://github.com/user-attachments/assets/3827d891-a8d1-4e8e-b424-444df0d05191" />

**Comment by gwilliams00** [comment]
Could a once-per-request version of this sit alongside #2869? The thing I actually want on a running show is a single "you're caught up" message: fire it the first time every episode that has aired so far is downloaded and there are still episodes to come. After that, nothing until the season finishes, when the normal available notification does its job.

The data is already there, the TMDB air dates and the per-season count that drives partially available. It's one comparison, so it could be on by default without the volume problem, and the per-episode notification from #2869 could stay an opt-in for people who want it.

Wording along the lines of the existing ones: "Ludwig (2024) Season 2 is now available. The season is currently airing; new episodes will follow."

A few edge cases I'd want covered: a request for several seasons counts as caught up once the finished seasons are complete and the airing one meets the rule; an episode with no air date yet is treated as unaired; season 0 is skipped the same way the scanners skip it unless specials are enabled.
